certificate.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2022 API7.ai, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloud
+
+import (
+	"context"
+	"path"
+	"time"
+)
+
+// Certificate is the definition of API7 Cloud Certificate.
+// some management fields.
+type Certificate struct {
+	CertificateSpec `json:",inline"`
+
+	// ID is the unique identify to mark an object.
+	ID ID `json:"id"`
+	// ControlPlaneID is id of control plane that current certificate belong with
+	ControlPlaneID ID `json:"control_plane_id"`
+	// Status is status of certificate
+	Status EntityStatus `json:"status"`
+	// CreatedAt is the object creation time.
+	CreatedAt time.Time `json:"created_at"`
+	// UpdatedAt is the last modified time of this object.
+	UpdatedAt time.Time `json:"updated_at"`
+}
+
+// CertificateSpec is the specification of the Certificate
+type CertificateSpec struct {
+	// The certificate in PEM format.
+	Certificate string `json:"certificate"`
+	// Private key is the private key for the corresponding certificate in PEM format.
+	PrivateKey string `json:"private_key"`
+	// CACertificate is the client ca certificate in PEM format used to verify client certificate.
+	CACertificate string `json:"ca_certificate,omitempty"`
+	// Labels are used for resource classification and indexing
+	Labels []string `json:"labels,omitempty"`
```

Comment:
Add comment for `Type`.

Also, show optional values for Type in the comment.

Reply:
> Add comment for `Type`.
> 
> Also, show optional values for Type in the comment.

Added comments.

Reply:
We don't use magic number of literal strings. Please refer to other constants.

Reply:
> 

I draw the type as a constant.